src/app/CommandInputField.tsx

```diff
@@ -24,6 +26,11 @@ export const CommandInputField = observer((props: { controller: BottomPanelContr
 
     if (e.key === "Enter") {
       e.preventDefault();
+      lastCmd.push(inputValue);
+      if (lastCmd[lastCmdIndex.get()] !== inputValue) {
+        lastCmdIndex.set(lastCmd.length);
```

Comment:
A problem with your solution, the following are the steps to reproduce:

Execute `A`
Execute `B`
Execute `C`
Execute `D`
Arrow Up (Show `D`)
Arrow Up (Show `C`)
Arrow Up (Show `B`)
Arrow Up (Show `A`)
Delete `A`
Type `Z`
Press Enter
Arrow Up (show `undefined`) <-- issue

Fix: you should write `lastCmdIndex.set(lastCmd.length - 1);` instead.

Reply:
I see!! Thank you very much!!!